action/protocol/generic_validator.go

```diff
@@ -58,7 +58,8 @@ func (v *GenericValidator) Validate(ctx context.Context, selp *action.SealedEnve
 	if caller == nil {
 		return errors.New("failed to get address")
 	}
-	if err = selp.Envelope.SanityCheck(); err != nil {
+	featureCtx := MustGetFeatureCtx(ctx)
+	if err = selp.Envelope.SanityCheck(); err != nil && !featureCtx.Tolerate(err) {
```

Comment:
`featureCtx.TolerateEmptyCandidateName && errors.Cause(err) == action.ErrInvalidCanName` shouldn't be hidden in a function of `FeatureCtx`.

Reply:
I see, I was thinking in the future may have similar case again, so add a `Tolerate` for this kind of special case